dev/cognitiveservices/data-plane/Language/generate-questionanswers.json

```diff
@@ -0,0 +1,643 @@
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/QnAGenerationRequestBody"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "202": {
+            "description": "A successful call results with an Operation-Location header used to check the status of the job.",
+            "headers": {
+              "Operation-Location": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/generate-questionanswers/SuccessfulSubmitQnAGenerationJob.json"
+          },
+          "Successful query with marked documents": {
+            "$ref": "./examples/generate-questionanswers/SuccessfulSubmitQnAGenerationJob_useAnswerMarkedInContent.json"
+          },
+          "Successful query with URL in a document": {
+            "$ref": "./examples/generate-questionanswers/SuccessfulSubmitQnAGenerationJob_URL.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/generate-questionanswers/jobs/{jobId}": {
+      "get": {
+        "summary": "Gets the status of a QnA generation job.",
+        "operationId": "GenerateQuestionAnswers_GetJobStatus",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/JobIdParameter"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "QnA generation job status.",
+            "schema": {
+              "$ref": "#/definitions/QnAGenerationJobState"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/generate-questionanswers/SuccessfulQnAGenerationJobStatus.json"
+          }
+        }
+      }
+    },
+    "/generate-questionanswers/jobs/{jobId}/result": {
```

Comment:
This is non-standard. If you're using an action LRO, the final result should be in the job details itself. See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#obtaining-status-and-results-of-long-running-operations. The result would be in a `result` property of the completed job status.

You're also missing a `DELETE` to cancel the job, or is that not supported? What about listing all jobs? The link above has all the details we want to see in 1 of 2 LRO patterns (the other being a resource LRO, which doesn't make sense here).

Reply:
Removed the non-standard API call for result. Added result property in JobStatus response.